src/frontend/js/components/AutocompleteTopicField/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react';
+import Autosuggest, { InputProps } from 'react-autosuggest';
+import { QueryFunction, QueryKey, useQueryClient } from 'react-query';
+import { differenceBy } from 'lodash';
+import { useAsyncEffect } from 'utils/useAsyncEffect';
+import { fetchList } from 'data/fetchList';
+import * as types from 'types';
+import { Nullable } from 'types/utils';
+import { useCurrentUser } from 'data/useCurrentUser';
+
+function calcdDepth(depth: number) {
+  let depthClass: string;
+
+  switch (depth) {
+    case 0:
+    case 1:
+      depthClass = '';
+      break;
+
+    case 2:
+      depthClass = 'pl-6';
+      break;
+
+    case 3:
+      depthClass = 'pl-10';
+      break;
+
+    default:
+      depthClass = 'pl-14';
+      break;
+  }
+  return depthClass;
+}
+
+interface AutocompleteTopicFieldProps {
+  filterSuggestions?: Nullable<(suggestions: types.Topic[]) => types.Topic[]>;
+  inputProps?: Partial<InputProps<types.Topic>>;
+  onSuggestionSelected: (suggestion: types.Topic) => void;
+}
+
+export const AutocompleteTopicField = ({
+  filterSuggestions,
+  inputProps = {},
+  onSuggestionSelected,
+}: AutocompleteTopicFieldProps) => {
+  const queryClient = useQueryClient();
+  const [value, setValue] = useState('');
+  const [suggestions, setSuggestions] = useState<types.Topic[]>([]);
+  const { currentUser } = useCurrentUser();
+
+  const getTopics: Autosuggest.SuggestionsFetchRequested = async ({
+    value,
+  }) => {
+    const topics: types.APIList<types.Topic> = await queryClient.fetchQuery(
+      ['topics', { query: value }],
+      fetchList as QueryFunction<any, QueryKey>,
+    );
+
+    const filteredTopics = topics.results.filter(
```

Comment:
I don't think we should filter topics here, we should just show all returned topics